lib/softaculous_installer.php

```diff
@@ -0,0 +1,133 @@
+<?php
+abstract class SoftactulousInstaller
+{
+    /**
+     * @var Monolog\Logger An instance of the logger
+     */
+    protected $logger;
+
+    /**
+     * The installer constructor
+     *
+     * @param Monolog\Logger $logger An instance of the logger
+     */
+    public function __construct($logger)
+    {
+        Loader::loadComponents($this, ['Input']);
+        $this->logger = $logger;
+    }
+
+    /**
+     * Validates informations and runs a softaculous installation script on the given web panel
+     *
+     * @param stdClass $service An object representing the web panel service to execute a script for
+     * @param stdClass $meta The module row meta data for the service
+     * @return boolean Whether the script succeeded
+     */
+    abstract public function install(stdClass $service, stdClass $meta);
+
+    /**
+     * Gets a list of scripts available in softaculous
+     *
+     * @global type $SoftaculousScripts
+     * @global type $add_SoftaculousScripts
+     * @return array A list of scripts available in softaculous
+     */
+    protected function softaculousScripts()
+    {
+        global $SoftaculousScripts, $add_SoftaculousScripts;
+
+        // Set the curl parameters.
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, 'https://api.softaculous.com/scripts.php?in=serialize');
+
+        // Turn off the server and peer verification (TrustManager Concept).
+        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+        curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);
+        curl_setopt($ch, CURLOPT_HEADER, false);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        //curl_setopt($ch, CURLOPT_SSL_CIPHER_LIST, 'RC4-SHA:RC4-MD5');
+        // This is becuase some servers cannot access https without this
+
+        // Get response from the server
+        $resp = curl_exec($ch);
+        $scripts = unserialize($resp);
```

Comment:
I would think this API request can be updated for JSON responses rather than serialized responses

Reply:
If I try it just returns a php file. But it is possible I may be able to pull in json  using a different url that would work better. I'll look into that

Reply:
Sounds good, we'd prefer not to do php unserialization.